src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+                                                yIntersect, chkDist);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_8003AB28.s")
+// Checks if point (`x`,`z`) is within `chkDist` of `poly`, computing `yIntersect` if true
+// Determinant max 0.0f (checks if on or within poly)
+s32 func_80038E78(CollisionPoly* poly, Vec3s* vtxList, f32 x, f32 z, f32* yIntersect, f32 chkDist) {
+    static Vec3f polyVtxs[3];
+    f32 nx;
+    f32 ny;
+    f32 nz;
+
+    CollisionPoly_GetVertices(poly, vtxList, polyVtxs);
+    CollisionPoly_GetNormal(poly, &nx, &ny, &nz);
+    return Math3D_TriChkPointParaYIntersectInsideTri(&polyVtxs[0], &polyVtxs[1], &polyVtxs[2], nx, ny, nz, poly->dist,
+                                                     z, x, yIntersect, chkDist);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_8003AC54.s")
+// Checks if point (`x`,`z`) is within 1.0f of `poly`, computing `yIntersect` if true
+// Determinant max 300.0f
+s32 func_80038F20(CollisionPoly* poly, Vec3s* vtxList, f32 x, f32 z, f32* yIntersect) {
+    return func_80038D48(poly, vtxList, x, z, yIntersect, 1.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_8003AD00.s")
+// Checks if point (`y`,`z`) is within 1.0f of `poly`, computing `xIntersect` if true
+// Determinant max 300.0f
+s32 func_80038F60(CollisionPoly* poly, Vec3s* vtxList, f32 y, f32 z, f32* xIntersect) {
+    static Vec3f polyVtxs[3];
+    f32 nx;
+    f32 ny;
+    f32 nz;
+
+    CollisionPoly_GetVertices(poly, vtxList, polyVtxs);
+    CollisionPoly_GetNormal(poly, &nx, &ny, &nz);
+    return Math3D_TriChkPointParaXIntersect(&polyVtxs[0], &polyVtxs[1], &polyVtxs[2], nx, ny, nz, poly->dist, y, z,
+                                            xIntersect);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_8003ADC8.s")
+// Checks if point (`x`,`y`) is within 1.0f of `poly`, computing `zIntersect` if true
+// Determinant max 300.0f
+s32 func_80039000(CollisionPoly* poly, Vec3s* vtxList, f32 x, f32 y, f32* zIntersect) {
+    static Vec3f polyVtxs[3];
+    f32 nx;
+    f32 ny;
+    f32 nz;
+
+    CollisionPoly_GetVertices(poly, vtxList, polyVtxs);
+    CollisionPoly_GetNormal(poly, &nx, &ny, &nz);
+    return Math3D_TriChkPointParaZIntersect(&polyVtxs[0], &polyVtxs[1], &polyVtxs[2], nx, ny, nz, poly->dist, x, y,
+                                            zIntersect);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_8003AEA8.s")
+// Test if travelling from `posA` to `posB` intersects `poly`
+// returns true if an intersection occurs, else false
+// returns `planeIntersect`, which is the point at which the line from `posA` to `posB` crosses `poly`'s plane
+// if `chkOneFace` is true, ignore negative to positive direction intersections
+s32 func_800390A0(CollisionPoly* poly, Vec3s* vtxList, Vec3f* posA, Vec3f* posB, Vec3f* planeIntersect, s32 chkOneFace,
```

Comment:
I think the var/flag name change suggestion is a bit overkill. I don't think it's super necessary to incorporate the check direction within the var/flag name because there isn't a way to enable checking against just the opposing face of the poly.